From 7e3178bee6749a479339ec8e79c6e7f912b97c17 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Mon, 10 Sep 2018 13:03:05 +0200
Subject: [PATCH 0683/1051] phy: cp110-comphy: introduce minor improvements to
 comphy driver

- fix formatting
- simplify some parts of the code
- use more verbose names for macros and reuse already defined ones
where possible

Change-Id: I728d3e6e639c1fefb9f0d730f1c633ff49d8a580
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/phy/marvell/phy-mvebu-cp110-comphy.c | 32 +++++++++-----------
 1 file changed, 15 insertions(+), 17 deletions(-)

diff --git a/drivers/phy/marvell/phy-mvebu-cp110-comphy.c b/drivers/phy/marvell/phy-mvebu-cp110-comphy.c
index cf720b640187..cf305e92e9ec 100644
--- a/drivers/phy/marvell/phy-mvebu-cp110-comphy.c
+++ b/drivers/phy/marvell/phy-mvebu-cp110-comphy.c
@@ -38,11 +38,11 @@ struct mvebu_comhy_conf {
 #define MV_SIP_COMPHY_DIG_RESET	0x82000005
 
 #define COMPHY_FW_MODE_FORMAT(mode)		((mode) << 12)
-#define COMPHY_FW_FORMAT(mode, idx, speeds)	\
-			(((mode) << 12) | ((idx) << 8) | ((speeds) << 2))
+#define COMPHY_FW_NET_FORMAT(mode, idx, speeds)	\
+		(COMPHY_FW_MODE_FORMAT(mode) | ((idx) << 8) | ((speeds) << 2))
 
 #define COMPHY_FW_PCIE_FORMAT(pcie_width, mode, idx, speeds)	\
-		    (((pcie_width) << 18) | COMPHY_FW_FORMAT(mode, idx, speeds))
+		(((pcie_width) << 18) | COMPHY_FW_NET_FORMAT(mode, idx, speeds))
 
 #define COMPHY_SATA_MODE	0x1
 #define COMPHY_SGMII_MODE	0x2	/* SGMII 1G */
@@ -66,7 +66,8 @@ struct mvebu_comhy_conf {
 #define COMPHY_SPEED_MAX		0x3F
 
 static unsigned long comphy_smc(unsigned long function_id,
-	   phys_addr_t comphy_phys_addr, unsigned long lane, unsigned long mode)
+				phys_addr_t comphy_phys_addr,
+				unsigned long lane, unsigned long mode)
 {
 	struct arm_smccc_res res;
 
@@ -153,25 +154,25 @@ static int mvebu_comphy_power_on(struct phy *phy)
 	case PHY_MODE_SGMII:
 		ret = comphy_smc(MV_SIP_COMPHY_POWER_ON, priv->phys,
 				 lane->id,
-				 COMPHY_FW_FORMAT(COMPHY_SGMII_MODE,
-						  lane->port,
-						  COMPHY_SPEED_1_25G));
+				 COMPHY_FW_NET_FORMAT(COMPHY_SGMII_MODE,
+						      lane->port,
+						      COMPHY_SPEED_1_25G));
 
 		break;
 	case PHY_MODE_2500SGMII:
 		ret = comphy_smc(MV_SIP_COMPHY_POWER_ON, priv->phys,
 				 lane->id,
-				 COMPHY_FW_FORMAT(COMPHY_HS_SGMII_MODE,
-						  lane->port,
-						  COMPHY_SPEED_3_125G));
+				 COMPHY_FW_NET_FORMAT(COMPHY_HS_SGMII_MODE,
+						      lane->port,
+						      COMPHY_SPEED_3_125G));
 
 		break;
 	case PHY_MODE_10GKR:
 		ret = comphy_smc(MV_SIP_COMPHY_POWER_ON, priv->phys,
 				 lane->id,
-				 COMPHY_FW_FORMAT(COMPHY_XFI_MODE,
-						  lane->port,
-						  COMPHY_SPEED_10_3125G));
+				 COMPHY_FW_NET_FORMAT(COMPHY_XFI_MODE,
+						      lane->port,
+						      COMPHY_SPEED_10_3125G));
 		break;
 	case PHY_MODE_PCIE:
 		ret = comphy_smc(MV_SIP_COMPHY_POWER_ON, priv->phys,
@@ -215,11 +216,8 @@ static int mvebu_comphy_power_off(struct phy *phy)
 {
 	struct mvebu_comphy_lane *lane = phy_get_drvdata(phy);
 	struct mvebu_comphy_priv *priv = lane->priv;
-	int ret;
-
-	ret = comphy_smc(MV_SIP_COMPHY_POWER_OFF, priv->phys, lane->id, 0);
 
-	return ret;
+	return comphy_smc(MV_SIP_COMPHY_POWER_OFF, priv->phys, lane->id, 0);
 }
 
 static const struct phy_ops mvebu_comphy_ops = {
-- 
2.17.1

